client/gutenberg/extensions/slideshow/create-swiper.js

```diff
@@ -31,17 +37,11 @@ export default async function createSwiper(
 		releaseFormElements: false,
 		setWrapperSize: true,
 		touchStartPreventDefault: false,
-		on: mapValues(
-			callbacks,
-			callback =>
-				function() {
-					callback( this );
-				}
-		),
+		on,
 	};
 	const [ { default: Swiper } ] = await Promise.all( [
 		import( /* webpackChunkName: "swiper" */ 'swiper' ),
 		import( /* webpackChunkName: "swiper" */ 'swiper/dist/css/swiper.css' ),
 	] );
-	return new Swiper( container, merge( {}, defaultParams, params ) );
+	return new Swiper( container, Object.assign( {}, defaultParams, params ) );
```

Comment:
Hmm, unlike `_.merge()`, `Object.assign()` isn't recursive -- it only merges the top-level elements of keys. (And unfortunately, [there is no `Object.merge()`](https://stackoverflow.com/questions/27936772/how-to-deep-merge-instead-of-shallow-merge) or anything like that in JS itself.)

I think that for properties of `params` that are objects themselves (such as `navigation` and `pagination`), this would mean that they would override their `defaultParams` counterpart in their entirety. E.g.

```js
const defaultParams = {
    effect: 'slide',
    pagination: {
        clickable: true,
        el: '.swiper-pagination',
        type: 'bullets',
    },
    /* ... */
}

const params = {
    effect: 'fade',
    pagination: {
        type: 'emoji',
    },
    /* ... */
}

merge( {}, defaultParams, params ) === {
    effect: 'fade',
    pagination: {
        clickable: true,
        el: '.swiper-pagination',
        type: 'emoji',
    },
    /* ... */
}

Object.assign( {}, defaultParams, params ) === {
    effect: 'fade',
    pagination: {
        type: 'emoji',
    },
    /* ... */
}
```

Meaning that a consumer would have to include the default values for nested (`navigation` and `pagination`) properties in `params` if they don't want to lose those.

We'd need a pretty big warning to avoid confusion here I guess :grimacing: -- it kind of undermines the concept of the default :slightly_frowning_face: since while things work as expected for top-level, 'flat' properties, they don't for nested ones...